crates/data_components/src/object/mod.rs

```diff
@@ -45,7 +45,32 @@ impl ObjectStoreContext {
     ) -> Result<Self, Box<dyn std::error::Error + Send + Sync>> {
         let (prefix, filename_regex_opt) = parse_prefix_and_regex(url, extension)?;
         let filename_regex = filename_regex_opt
-            .map(|regex| Regex::new(&regex).boxed())
+            .map(|regex| {
+                // Prevent ReDoS attacks by limiting regex complexity
+                // Length check catches obviously malicious patterns
+                const MAX_REGEX_LENGTH: usize = 1000;
+                if regex.len() > MAX_REGEX_LENGTH {
+                    return Err(format!(
+                        "Regex pattern too long ({} chars). Maximum allowed: {MAX_REGEX_LENGTH}",
+                        regex.len()
+                    )
+                    .into());
+                }
+
+                // Additional ReDoS protection: detect dangerous patterns
```

Comment:
If this is for glob support, use the glob crate's pattern struct: https://docs.rs/glob/latest/glob/struct.Pattern.html

It already has some validations for acceptable patterns which should be more restrictive than for an arbitrary regex